src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/RequestFailureReason.java

```diff
@@ -0,0 +1,62 @@
+package io.stargate.sgv2.jsonapi.service.cqldriver.executor;
+
+/**
+ * Roughly copied from `RequestFailureReason` in the Cassandra code base, but simplified to only the
+ * codes.
+ *
+ * <p><b>NOTE:</b> The codes here need to be in sync with the C* code, the exceptions in the drivers
+ * only use the codes. Sync with <em>Astra</em> version of the code not the DataStax because the
+ * Astra version has more codes.
+ *
+ * <p>Used with {@link com.datastax.oss.driver.api.core.servererrors.ReadFailureException} and
+ * similar errors.
+ */
+public enum RequestFailureReason {
+  // codes must be sequential !
```

Comment:
Do we ever need to worry about 500+ codes that seem to exist for DataStax-DBs (DSE and/or HCD)?

Reply:
not for this, it's internal C* codes that writers / worker nodes return to the coordinator for why something failed. Which are then passed back to the client. 

See commend at top they are copied from RequestFailureReason in the C* code. This is where we can detect that an index is not available and give correct error